packages/flutter/lib/src/services/haptic_feedback.dart

```diff
@@ -80,4 +82,18 @@ class HapticFeedback {
       'HapticFeedbackType.selectionClick',
     );
   }
+
+  @visibleForTesting
+  // ignore: public_member_api_docs
+  static const String isHapticEnabledMethodName = 'HapticFeedback.isEnabled';
+
+  /// Provides value for haptic feedback user setting.
+  ///
+  /// {@code false} indicates the user has opted out of haptic feedback.
+  /// If user preference is unknown or unknowable defaults to {@code true}.
+  static Future<bool?> isHapticFeedbackEnabled() async {
+    return SystemChannels.platform.invokeMethod<bool>(
```

Comment:
It is weird that we call this through the message channel. I remember changing most of the system settings will cause all activity to restart unless they set a flag in android manifest. If that is the case, the value would not change after activity is launched